realm-transformer/src/main/kotlin/io/realm/transformer/RealmTransformer.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.transformer
+
+import com.android.build.api.transform.*
+import javassist.CtClass
+import org.gradle.api.Project
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.File
+
+// Package level logger
+val logger: Logger = LoggerFactory.getLogger("realm-logger")
+
+/**
+ * This class implements the Transform API provided by the Android Gradle plugin.
+ */
+class RealmTransformer(val project: Project) : Transform() {
+
+    val logger: Logger = LoggerFactory.getLogger("realm-logger")
+
+    override fun getName(): String {
+        return "RealmTransformer"
+    }
+
+    override fun getInputTypes(): Set<QualifiedContent.ContentType> {
+        return setOf<QualifiedContent.ContentType>(QualifiedContent.DefaultContentType.CLASSES)
+    }
+
+    override fun isIncremental(): Boolean {
+        return true
+    }
+
+    override fun getScopes(): MutableSet<in QualifiedContent.Scope> {
+        return mutableSetOf(QualifiedContent.Scope.PROJECT)
```

Comment:
Because the interface is from Java, so in Kotlin you cannot specify `getScopes(): Set<>` due to immutable collections being a different type than `MutableSet`